packages/purpose-process/src/services/purposeService.ts

```diff
@@ -338,8 +339,12 @@ export function purposeServiceBuilder(
         purposeId: PurposeId;
         versionId: PurposeVersionId;
       },
-      { correlationId, authData, logger }: WithLogger<AppContext<UIAuthData>>
-    ): Promise<void> {
+      {
+        correlationId,
+        authData,
+        logger,
+      }: WithLogger<AppContext<UIAuthData | M2MAdminAuthData>>
+    ): Promise<{ metadata: Metadata }> {
```

Comment:
I know it's unused, but I would still return `WithMetadata<Purpose>` here, since the metadata itself makes sense only together with some data (the resource).

Also, could you update the corresponding integration test to check the new return value of the service call?